src/sources/hydra.rs

```diff
@@ -172,7 +198,20 @@ impl gasket::framework::Worker<Stage> for Worker {
             Message::Text(text) => {
                 debug!("Hydra message: {}", text);
                 match serde_json::from_str::<HydraMessage>(text) {
-                    Ok(hydra_message) => self.process_next(stage, hydra_message).await,
+                    Ok(hydra_message) => {
+                        // TODO: search for the intersection point to ensure
+                        // we're on the same chain.
+                        let should_process = match &self.intersect {
+                            Point::Origin => true,
+                            Point::Specific(slot,_hash) => &hydra_message.seq > slot
+                        };
+                        if should_process {
+                            self.process_next(stage, hydra_message).await
+                        } else {
+                            debug!("Skipping message before intersection");
```

Comment:
maybe would be nice to have more information here like. we are at XXX and set intersection is YYY, so skipping ...

Reply:
Done 4a1d0ca81e4daf24b57881970d9d4995eafd4817